### PR TITLE
Enhancement: Redesigning UI Components for Improved User Experience

### PR DESCRIPTION
**Pull request by mahmoud-eslami**
### Description
Hi! This PR aims to enhance the user interface by implementing a comprehensive redesign. 

### Tasks
- [x] Add native Loading component before load the main content.
- [x] Redesign the web version.
- [x] Add a GitHub icon and link it to the repository for easier access.
- [x] Search bar.

### CODE REVIEW

**Comment by mahmoud-eslami** [comment]
Hey @payam-zahedi, could you take a look and share your thoughts? I'm considering a black and white theme for the website, and I believe this style of splash screen might look beautiful in line with that theme.
The primary aim of this loading widget is to enhance user experience by displaying it instead of a blank screen when users open the website.

https://github.com/PersianFlutter/iconsax_flutter/assets/57481226/cf541187-eb1e-42f6-a7f1-4ff8afaae7bc

**Comment by payam-zahedi** [comment]
Awesome work @mahmoud-eslami 

it looks beautiful

File: android/local.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+sdk.dir=/Users/mahmoud/Library/Android/sdk
```
Comment (payam-zahedi) [inline android/local.properties:1]: I guess you should ignore this file in the git

File: example/ios/Podfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Uncomment this line to define a global platform for your project
```
Comment (payam-zahedi) [inline example/ios/Podfile]: Also we don't need ios and android folders for the example project

what do you think?

File: example/lib/main.dart [hunk 3]
```diff
@@ -13,121 +17,13 @@ class MyApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Iconsax Flutter',
+      title: 'Flutter Iconsax',
```
Comment (payam-zahedi) [inline example/lib/main.dart]: I think Iconsax Flutter is better because this is the package name

File: example/pubspec.lock [hunk 4]
```diff
@@ -37,10 +37,18 @@ packages:
     dependency: transitive
     description:
       name: collection
-      sha256: "4a07be6cb69c84d677a6c3096fcf960cc3285a8330b4603e0d463d15d9bd934c"
+      sha256: ee67cb0715911d28db6bf4af1026078bd6f0128b07a5f66fb2ed94ec6783c09a
```
Comment (payam-zahedi) [inline example/pubspec.lock]: Please also ignore the pubspec.lock in the git

File: example/pubspec.yaml [hunk 5]
```diff
@@ -12,8 +12,11 @@ dependencies:
     sdk: flutter
   iconsax_flutter:
     path: ../
-  url_launcher: ^6.1.12
-
+  url_launcher: ^6.2.2
+  google_fonts: ^6.1.0
+  hooks_riverpod:
+  flutter_hooks:
+  flutter_riverpod:
```
Comment (payam-zahedi) [inline example/pubspec.yaml]: Add versions plz otherwise we will have conflicts

File: ios/Flutter/Generated.xcconfig [hunk 6]
```diff
@@ -0,0 +1,14 @@
+// This is a generated file; do not edit or check into version control.
+FLUTTER_ROOT=/Users/mahmoud/.puro/envs/stable/flutter
+FLUTTER_APPLICATION_PATH=/Users/mahmoud/dev/projects/flutter/iconsax_flutter
+COCOAPODS_PARALLEL_CODE_SIGN=true
+FLUTTER_TARGET=lib/main.dart
+FLUTTER_BUILD_DIR=build
+FLUTTER_BUILD_NAME=1.0.0
+FLUTTER_BUILD_NUMBER=1.0.0
```
Comment (payam-zahedi) [inline ios/Flutter/Generated.xcconfig:8]: also, this file is personal plz ignore it

File: ios/Flutter/flutter_export_environment.sh [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+# This is a generated file; do not edit or check into version control.
```
Comment (payam-zahedi) [inline ios/Flutter/flutter_export_environment.sh:2]: also this one is personal . ignore it

**Comment by mahmoud-eslami** [comment]
@payam-zahedi Thanks for the review, I will fix all of them in the next commit.

File: example/ios/Podfile (hunk 2 above)
Comment (mahmoud-eslami) [reply example/ios/Podfile]: Yeah, I agree, in the current version we just support web, so I will delete other unused platforms.

**Comment by mahmoud-eslami** [comment]
Hey @payam-zahedi, this is the final UI, what is your opinion about it? 


https://github.com/PersianFlutter/iconsax_flutter/assets/57481226/fabe35a1-948c-45fc-8411-66050433e82f

**Comment by payam-zahedi** [comment]
Well Done @mahmoud-eslami 

It looks really good

is it ready for merging?

**Comment by mahmoud-eslami** [comment]
@payam-zahedi Thanks!
Yes it is, there are other changes, such as updating the README or deployment, which I'll handle in a separate PR.

**Comment by mahmoud-eslami** [comment]
Now is better 😅, I removed unused comment and print.

File: android/local.properties (hunk 1 above)
Comment (payam-zahedi) [reply android/local.properties:1]: You should delete this file from the git cache 

take a look at this answer 
https://stackoverflow.com/a/25436481

**Review by payam-zahedi — APPROVED** [review]
Everything is fine @mahmoud-eslami 
Thanks for your contribution, you are now part of Persian flutter community.

**Comment by payam-zahedi** [comment]
@mahmoud-eslami Just keep in mind to remote the android/local.properties file in the future PR.

**Comment by mahmoud-eslami** [comment]
@payam-zahedi Ohh, These files are in the root of the project, I thought that those are in the example folder and will remove after clean android and IOS folders from ./example directory. My bad, I will delete them in the next PR.

**Pull request merged**